### PR TITLE
Add assignId API for AnyCellmlElement.

### LINKED ISSUE #879: Add assignId(AnyCellmlElement) to Annotator
The `Annotator` class could do with having `assignId(AnyCellmlElement)` API added.

### PR DESCRIPTION
**Pull request by hsorby**
Fixes #879.

### CODE REVIEW

**Comment by hsorby** [comment]
This should be an easy one???  It is ready for review.

File: src/api/libcellml/annotator.h [hunk 1]
```diff
@@ -652,6 +652,27 @@ class LIBCELLML_EXPORT Annotator: public Logger
...
+     * @brief Assign an automatically generated, unique identifier to the given @p item.
+     *
+     * Assign an automatically generated, unique identifier to the given @p item.
+     * This method will return the new identifier that has been assigned, or an empty string
+     * if the operation failed.
+     *
+     * An identifier will not be assigned if:
+     *  - no model has been stored in this annotator;
+     *  - the given @p item is not a member of the stored model;
+     *  - the given @p item is not relevant to the given @p type; or
```
Comment (agarny) [inline src/api/libcellml/annotator.h]: Delete since no type here.

File: src/api/libcellml/annotator.h [hunk 2]
```diff
@@ -652,6 +652,27 @@ class LIBCELLML_EXPORT Annotator: public Logger
...
+     *  - the given @p item is @c nullptr.
+     *
+     * @see errorCount() if and/or error(size_t) for any issues that may have been raised.
```
Comment (agarny) [inline src/api/libcellml/annotator.h]: "error(size_t)"? Hmm... not sure we want that kind of description. People should have to know what `size_t` is about.

Comment (hsorby) [reply src/api/libcellml/annotator.h]: I think this should be `@see error(size_t)` so that doxygen can find the method.

**Review by nickerso — APPROVED** [review]

**Review by agarny — APPROVED** [review]

**Pull request merged**